modules/apps/collaboration/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/servlet/taglib/GroupSelectorTag.java

```diff
@@ -95,9 +96,20 @@ protected void cleanUp() {
 
 		groupParams.put("site", Boolean.TRUE);
 
-		return GroupLocalServiceUtil.search(
-			themeDisplay.getCompanyId(), _CLASSNAME_IDS, keywords,
-			groupParams, startAndEnd[0], startAndEnd[1], null);
+		if (_checkPermission(themeDisplay.getPermissionChecker())) {
+			return GroupLocalServiceUtil.search(
+				themeDisplay.getCompanyId(), _CLASSNAME_IDS, keywords,
+				groupParams, startAndEnd[0], startAndEnd[1], null);
+		}
+		else {
+			return GroupLocalServiceUtil.search(
+				themeDisplay.getCompanyId(),
+				new long[] {
+					ClassNameLocalServiceUtil.getClassNameId(Group.class),
```

Comment:
could you create a new constant with this classNames in order to improve readability?

I suggest change current _CLASSNAME_IDS constant to _COMPANY_ADMIN_CLASSNAME_IDS and create a new constant (with group and Organization classnameid ) as _CLASSNAME_IDS

Are you ok with this changes?